File: Rotation_Transformation.m

% Rotation Transformation
clc; clear; close all;

square = [0 0; 50 0; 50 50; 0 50; 0 0];
theta = 45 * pi/180;

R = [cos(theta) -sin(theta); sin(theta) cos(theta)];
rotated = (R * square')';

plot(square(:,1), square(:,2));
hold on;
plot(rotated(:,1), rotated(:,2));
axis equal; grid on;
title('Rotation Transformation (45 Degrees)');
legend('Original','Rotated');

File: Rotation_Transformation.py

# Rotation Transformation
import numpy as np
import matplotlib.pyplot as plt

square = np.array([[0,0],[50,0],[50,50],[0,50],[0,0]])
theta = np.deg2rad(45)

R = np.array([[np.cos(theta), -np.sin(theta)],
              [np.sin(theta),  np.cos(theta)]])

rotated = square.dot(R.T)

plt.plot(square[:,0], square[:,1])
plt.plot(rotated[:,0], rotated[:,1])
plt.gca().set_aspect('equal', 'box')
plt.title("Rotation Transformation (45 Degrees)")
plt.legend(["Original","Rotated"])
plt.grid(True)
plt.show()
